osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,341 @@
+                            },
+                            new BeatmapSetOnlineStatusPill
+                            {
+                                AutoSizeAxes = Axes.Both,
+                                Anchor = Anchor.TopRight,
+                                Origin = Anchor.TopRight,
+                                Shear = -wedged_container_shear,
+                                TextSize = 11,
+                                TextPadding = new MarginPadding { Horizontal = 8, Vertical = 2 },
+                                Status = beatmapInfo.Status,
+                                Alpha = string.IsNullOrEmpty(beatmapInfo.DifficultyName) ? 0 : 1
+                            }
+                        }
+                    },
+                    new FillFlowContainer
+                    {
+                        Name = "Top-left aligned metadata",
+                        Direction = FillDirection.Vertical,
+                        Position = new Vector2(text_margin + shear_width, 12),
+                        Width = .7f,
+                        AutoSizeAxes = Axes.Y,
+                        RelativeSizeAxes = Axes.X,
+                        Children = new Drawable[]
+                        {
+                            TitleLabel = new OsuSpriteText
+                            {
+                                Shadow = true,
+                                Current = { BindTarget = titleBinding },
+                                Font = OsuFont.TorusAlternate.With(size: 40, weight: FontWeight.SemiBold),
+                                RelativeSizeAxes = Axes.X,
+                                Truncate = true
+                            },
+                            ArtistLabel = new OsuSpriteText
+                            {
+                                // TODO : figma design has a diffused shadow, instead of the solid one present here, not possible currently as far as i'm aware.
+                                Shadow = true,
+                                Current = { BindTarget = artistBinding },
+                                // Not sure if this should be semi bold or medium
+                                Font = OsuFont.Torus.With(size: 20, weight: FontWeight.SemiBold),
+                                RelativeSizeAxes = Axes.X,
+                                Truncate = true
+                            }
+                        }
+                    }
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                starRatingDisplay.DisplayedStars.BindValueChanged(s =>
+                {
+                    wedge.starCounter.Current = (float)s.NewValue;
+                    wedge.starCounter.Colour = s.NewValue >= 6.5 ? colours.Orange1 : Colour4.Black.Opacity(0.75f);
+
+                    wedge.difficultyColourBar.FadeColour(colours.ForStarDifficulty(s.NewValue));
+                }, true);
```

Comment:
Not sure this is worth caching wedge for.

Reply:
Definitely feels back-to-front. I'd expose this value and then consume it from the wedge.